### DISCUSSION TITLE
Is it possible to automatically link static FMT or even disable it?

### DISCUSSION THREAD

**Discussion by dwarfovich** [gabime/spdlog#2442] [Q&A] [answered]
I just tested spdlog as static library (it's installed via vcpkg, using x86-windows-static triplet). But MSVS automatically adds fmt.dll to my project's build folder because it linked dynamically. I want to compile completely static project. So, how to link statically against FMT, should I do it myself? Or, could I completely remove it as a dependency?

**Comment by tt4g** [comment, marked as answer]
Unless the CMake variable `SPDLOG_BUILD_SHARED` (or `BUILD_SHARED_LIBS`) is defined, spdlog is built as a static library, so the bundled fmt is also part of the library.
If you do not use the bundled fmt, you are responsible for providing the static library fmt.

**Reply by tt4g** [reply]
It appears that vcpkg is configured to use external fmt. This project does not maintenance vcpkg, so please ask in the [vcpkg project](https://github.com/microsoft/vcpkg).
